src/bpf_verifier.rs

```diff
@@ -0,0 +1,314 @@
+            ebpf::ARSH64_IMM => {}
+            ebpf::ARSH64_REG => {}
+
+            // BPF_JMP class
+            ebpf::JA => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JEQ_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JEQ_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JGT_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JGT_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JGE_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JGE_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JLT_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JLT_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JLE_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JLE_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSET_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSET_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JNE_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JNE_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSGT_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSGT_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSGE_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSGE_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSLT_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSLT_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSLE_IMM => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::JSLE_REG => {
+                verify_jmp_offset(prog, insn_ptr);
+            }
+            ebpf::CALL => {}
+            ebpf::TAIL_CALL => unimplemented!(),
+            ebpf::EXIT => {}
+
+            _ => {
+                panic!(
+                    "[Verifier] Error: unknown eBPF opcode {:#2x} (insn #{:?})",
+                    insn.opc, insn_ptr
+                );
+            }
+        }
+
+        verify_registers(&insn, store, insn_ptr);
+
+        insn_ptr += 1;
+    }
+
+    // insn_ptr should now be equal to number of instructions.
+    if insn_ptr != prog.len() / ebpf::INSN_SIZE {
+        panic!("[Verifier] Error: jumped out of code to #{:?}", insn_ptr);
```

Comment:
Return false instead of panicking?

Reply:
See comment about returning `Result<>`.  rbpf currently ignores return value